### PR TITLE
PUBDEV-7508: Update all R examples to use Tidyverse style guide

### PR DESCRIPTION
**Pull request by hannah-tillman**
These changes were all in the R Reference Guide and the User Guide. Please let me know if there are any other files. 

Also, please let me know if you want the test updates in this PR as well or if you would like them on a future ticket. 

https://0xdata.atlassian.net/browse/PUBDEV-7508

### CODE REVIEW

**Comment by ledell** [comment]
This look good!  I have some other suggestions on how to futher conform our R examples to the Tidyverse style guide (referenced in the JIRA if you want to see the whole thing).  In summary, here are a few changes that we still need to conform to better style:

- Add space around equals sign, ` = `, instead of `x=x`, do `x = x`.
- Add space for variable assignment (e.g. `b <- a` instead of `b<-a` or `b = a`.)
- Add space around variables, commas, etc
- Change any camelCase variable name to snake case, e.g. `irisPath` to `iris_path`
- Numbers should have a zero in front, `.8` should be `0.8`.

You could do those in this PR or a follow-up.

**Comment by angela0xdata** [comment]
@hannah-tillman the User Guide changes look good. a gradle issue is preventing me from being able to test the R docs to see if anything errors (though I don't expect any). We might have to wait until that is fixed, and then updated this branch.

**Comment by hannah-tillman** [comment]
> @hannah-tillman the User Guide changes look good. A gradle issue is preventing me from being able to test the R docs to see if anything errors (though I don't expect any). We might have to wait until that is fixed, and then updated this branch.

All good! I still have a lot more files to comb through after finishing the Steam updates :)

File: h2o-docs/src/product/data-munging/target-encoding.rst [hunk 1]
```diff
@@ -103,8 +103,8 @@ In this example, we will be trying to predict ``survived`` using the popular tit
     titanic[response] <- as.factor(titanic[response])
 
     # Split the dataset into train and test
-    seed=1234
-    splits <- h2o.splitFrame(titanic, seed = seed, ratios = c(0.8), destination_frames = c("train", "test"))
+    seed = 1234
```
Comment (ledell) [inline h2o-docs/src/product/data-munging/target-encoding.rst:106]:
```suggestion
    seed <- 1234
```

**Comment by ledell** [comment]
@hannah-tillman This is a huge improvement, thanks!  I am going to update the JIRA ticket to reflect that we are making changes to match the tidyverse style guide.

File: h2o-r/h2o-package/R/gam.R [hunk 2]
```diff
@@ -117,7 +117,7 @@
 #' prostate <- h2o.uploadFile(path = prostate_path)
 #' prostate$CAPSULE <- as.factor(prostate$CAPSULE)
 #' h2o.gam(y = "CAPSULE", x = c("RACE"), gam_columns = c("PSA"),
-#'      training_frame = prostate,family = "binomial")
+#'      training_frame = prostate, family = "binomial")
```
Comment (angela0xdata) [inline h2o-r/h2o-package/R/gam.R:120]: @hannah-tillman this is the wrong file to update. The changes to this and other algos have to be made in:
/h2o-3/h2o-bindings/bin/custom/R

File: h2o-r/h2o-package/R/gbm.R [hunk 3]
```diff
@@ -111,7 +111,7 @@
 #' # Run regression GBM on australia data
 #' australia_path <- system.file("extdata", "australia.csv", package = "h2o")
 #' australia <- h2o.uploadFile(path = australia_path)
-#' independent <- c("premax", "salmax","minairtemp", "maxairtemp", "maxsst",
```
Comment (angela0xdata) [inline h2o-r/h2o-package/R/gbm.R:114]: This is the wrong file to update. The changes to this and other algos have to be made in:
/h2o-3/h2o-bindings/bin/custom/R

File: h2o-r/h2o-package/R/glm.R [hunk 4]
```diff
@@ -129,7 +129,7 @@
 #' # Run GLM of CAPSULE ~ AGE + RACE + PSA + DCAPS
 #' prostate_path = system.file("extdata", "prostate.csv", package = "h2o")
 #' prostate = h2o.importFile(path = prostate_path)
-#' h2o.glm(y = "CAPSULE", x = c("AGE","RACE","PSA","DCAPS"), training_frame = prostate,
```
Comment (angela0xdata) [inline h2o-r/h2o-package/R/glm.R:132]: This is the wrong file to update. The changes to this and other algos have to be made in:
/h2o-3/h2o-bindings/bin/custom/R

File: h2o-r/h2o-package/R/isolationforest.R [hunk 5]
```diff
@@ -48,7 +48,7 @@
 #' cars <- h2o.importFile(f)
 #' 
 #' # Set the predictors
-#' predictors <- c("displacement","power","weight","acceleration","year")
```
Comment (angela0xdata) [inline h2o-r/h2o-package/R/isolationforest.R:51]: This is the wrong file to update. The changes to this and other algos have to be made in:
/h2o-3/h2o-bindings/bin/custom/R

File: h2o-r/h2o-package/R/gam.R (hunk 2 above)
Comment (ledell) [reply h2o-r/h2o-package/R/gam.R:120]: Oh man... I have done this so many times... Sorry @hannah-tillman.  Thanks @angela0xdata for pointing this out. :-(

File: h2o-bindings/bin/custom/R/gen_word2vec.py [hunk 6]
```diff
@@ -22,12 +22,12 @@
 
 # Import the CraigslistJobTitles dataset
 f <- "https://raw.githubusercontent.com/h2oai/sparkling-water/rel-1.6/examples/smalldata/"
-jobtitles <- h2o.importFile(paste0(f, "craigslistJobTitles.csv"), 
-                             col.names = c("category", "jobtitle"),
+job_titles <- h2o.importFile(paste0(f, "craigslistJobTitles.csv"), 
+                             col.names = c("category", "job_title"),
```
Comment (angela0xdata) [inline h2o-bindings/bin/custom/R/gen_word2vec.py:26]: >col.names = c("category", "job_title")

This change is not correct and should be reverted. So what's happening is, the craigslistJobTitles.csv dataset includes two columns with headers: one named category and one named jobtitle. Your change here causes a new header row to be created and pushes the original header row into the data (see below).

```
> job_titles
   category                                            job_title
1  category                                             jobtitle
2 education                              After School Supervisor
3 education *****TUTORS NEEDED - FOR ALL SUBJECTS, ALL AGES***** 
```

**Pull request merged**